### PR TITLE
feature/OT221-62

### PR DESCRIPTION
**Pull request by rodrigovaldivia-dev**
## Issue ticket link
https://alkemy-labs.atlassian.net/browse/OT221-62

## Summary
-Added dynamic menu whether the user has an admin role or user role
-Shows admin menu of options or standard user menu
-Improves can be done once the testing branch gets merged
-Added menu button for users and admin

## Evidence
-If the user has a role of 1 the page will go directly to an admin page or to the outlet
2022-07-10 20_18_26-SOMOS MAS

-If the user doesn't meet requirements but has a role of 2 the site will redirect to the user profile path
2022-07-10 20_18_49-SOMOS MAS

-In the event that nobody is authenticated but tries to access a private path, it will be redirected to the login page 
2022-07-02 17_52_43-React App

-Menu button that navigates you to backoffice or user profile 
2022-07-11 22_35_17-SOMOS MAS

### CODE REVIEW

File: src/router/AppRouter.js [hunk 1]
```diff
@@ -19,13 +25,11 @@ const AppRouter = () => {
                             <DashboardRoutes />
                         </PrivateRoutes>
                     } />
-                    <Route path="/backoffice/*" element={
-                      /*   <PrivateRoutes> */ //no esta envuelt oya que esta private router solo dice si es usuario, no si es adminitrador
-                      // se deberia crear un componente que verifica eso y envuelva a el componente BackofficeRoutes
+                    <Route path="/backoffice/*" element={
```
Comment (GauntStrelok) [inline src/router/AppRouter.js]: Todas las rutas del backoffice requieren que vos seas admin?

File: src/routes/RequireAuthRole.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { useLocation, Navigate, Outlet } from "react-router-dom";
+
+
+const RequireAuthRole = ({ allowedRoles }) => {
+  const location = useLocation();
+
+  //TODO Here we should grab the user data from redux store... hardcoding user for now 
+  const auth = {
```
Comment (GauntStrelok) [inline src/routes/RequireAuthRole.js]: El estado de redux con el usuario no esta creado ya?

**Review by GauntStrelok — COMMENTED** [review]
Respondeme el comentario

File: src/routes/RequireAuthRole.js (hunk 2 above)
Comment (rodrigovaldivia-dev) [reply src/routes/RequireAuthRole.js]: Devuelve únicamente el token el store, uno de los chicos me dijo q estaba haciendo para que mando el rol y datos de usuario.

File: src/pages/Backoffice/MenuByRole.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import { useLocation, Navigate} from "react-router-dom";
+import { BackOffice } from "./BackOffice";
+
+
+const MenuByRole = () => {
+  const user = useSelector((state) => state.userLogged.user.user)
+  console.log(user)
```
Comment (GauntStrelok) [inline src/pages/Backoffice/MenuByRole.js]: Acordate de sacarle los console log

**Comment by GauntStrelok** [comment]
Hacer un pull de main para arreglar conflictos

**Pull request merged**